invenio_config_tugraz/generators.py

```diff
@@ -153,10 +153,76 @@ class ExampleRecordPermissionPolicy(BasePermissionPolicy):
 
 from elasticsearch_dsl.query import Q
 from flask import current_app, request
-from invenio_access.permissions import any_user, superuser_access
+from flask_principal import RoleNeed
+from invenio_access.permissions import any_user
+from invenio_accounts.models import User
 from invenio_records_permissions.generators import Generator
 
 
+class TUGrazCurators(Generator):
+    """Allows users with certain role."""
+
+    def __init__(self):
+        """Constructor."""
+        super(TUGrazCurators, self).__init__()
+
+    def get_curators(self):
+        """Get curators from config."""
+        return current_app.config.get("TUGRAZ_CURATORS", {})
+
+    def needs(self, record=None, **kwargs):
+        """Enabling Needs."""
+        if record is None:
+            return []
+
+        owners = record.parent.access.owned_by
+        # owners is empty for demo records. not sure about other records.
+        if len(owners) == 0:
+            return []
+
+        try:
+            owner = owners[0].resolve()
+        except Exception as e:
+            print(f"exception during owner lookup: {e}")
+            return []
+
+        curators = self.get_curators()
+        curator = None
+        for c in curators.values():
```

Comment:
does this mean, that the curator has to be a owner of the record. therefore it has to be added to the owner-field of the record?

Reply:
as I understood, the curator accounts will fetch external records and create them in our repository. this will make the curator account the owner of the record. other users with a curator role can then access the accounts created by the curator account